src/test/java/uk/gov/pay/api/service/AgreementServiceTest.java

```diff
@@ -53,40 +54,119 @@ public void setup() {
-        CreateAgreementResponse agreementResponse = agreementService.create(account, requestPayload);
-        assertThat(agreementResponse.getAgreementId(), is(agreementId));
-        assertThat(agreementResponse.getAgreementType(), is(AgreementType.ON_DEMAND));
-        assertThat(agreementResponse.getCreatedDate(), is("2016-01-01T12:00:00.000Z"));
-        assertThat(agreementResponse.getReturnUrl(), is("https://example.com/return"));
-        assertThat(agreementResponse.getState(), is(AgreementStatus.CREATED));
-        assertThat(agreementResponse.getLinks().getSelf(), is(new Link("https://publicapi.test/v1/agreements/agreementid", "GET")));
-        assertThat(agreementResponse.getLinks().getNextUrl(), is(new Link("http://frontend_direct_debit/secure/token_1234567asdf", "GET")));
-        PostLink expectedLink = new PostLink("http://frontend_direct_debit/secure/", "POST", "application/x-www-form-urlencoded", Collections.singletonMap("chargeTokenId", "token_1234567asdf"));
-        assertThat(agreementResponse.getLinks().getNextUrlPost(), is(expectedLink));
+        MandateConnectorRequest mandateConnectorRequest = new MandateConnectorRequest(
+                "https://example.com/return",
+                AgreementType.ON_DEMAND,
+                null
+        );
+        Response connectorResponse = agreementService.createMandate(account, mandateConnectorRequest);
+        MandateConnectorResponse mandateConnectorResponse = connectorResponse.readEntity(MandateConnectorResponse.class);
+
+        assertThat(mandateConnectorResponse.getMandateId(), is(MANDATE_ID));
+        assertThat(mandateConnectorResponse.getMandateType(), is(MandateType.ON_DEMAND));
+        assertThat(mandateConnectorResponse.getCreatedDate(), is("2016-01-01T12:00:00.000Z"));
+        assertThat(mandateConnectorResponse.getReturnUrl(), is("https://example.com/return"));
+        assertThat(mandateConnectorResponse.getState(), is(new MandateState("created", false)));
+        assertThat(mandateConnectorResponse.getLinks().get(0), is(new PaymentConnectorResponseLink(
+                "self",
+                "http://localhost:1234/v1/api/accounts/9ddfcc27-acf5-43f9-92d5-52247540714b/mandates/" + MANDATE_ID,
+                "GET",
+                null,
+                null
+        )));
+        assertThat(mandateConnectorResponse.getLinks().get(1), is(new PaymentConnectorResponseLink(
+                "next_url",
+                "http://frontend_direct_debit/secure/token_1234567asdf",
+                "GET",
+                null,
+                null
+        )));
+        assertThat(mandateConnectorResponse.getLinks().get(2), is(new PaymentConnectorResponseLink(
+                "next_url_post",
+                "http://frontend_direct_debit/secure/",
+                "POST",
+                "application/x-www-form-urlencoded",
+                Collections.singletonMap("chargeTokenId", "token_1234567asdf")
+        )));
```

Comment:
are we missing an assertion for serviceReference here? should it be null/empty? or will it be generated?

Reply:
Well spotted!